pkg/controller/uniteddeployment/uniteddeployment_update.go

```diff
@@ -30,6 +30,49 @@ import (
 	"github.com/openkruise/kruise/pkg/util"
 )
 
+func (r *ReconcileUnitedDeployment) manageSubsets(ud *appsv1alpha1.UnitedDeployment, nameToSubset *map[string]*Subset, nextReplicas, nextPartitions *map[string]int32, currentRevision, updatedRevision *appsv1.ControllerRevision, subsetType subSetType) (updateErr error) {
+	exists, err := r.manageSubsetProvision(ud, nameToSubset, nextReplicas, nextPartitions, currentRevision, updatedRevision, subsetType)
+	if err != nil {
+		return fmt.Errorf("fail to manage Subset provision: %s", err)
+	}
+
+	expectedRevision := currentRevision
+	if updatedRevision != nil {
+		expectedRevision = updatedRevision
+	}
+
+	var needUpdate []string
+	for _, name := range exists.List() {
+		subset := (*nameToSubset)[name]
+		expectedPartition := (*nextPartitions)[name]
```

Comment:
Nit: we may not need expectedPartition variable as it is only used once.

Reply:
done